src/CacheEntry.php

```diff
@@ -256,47 +257,72 @@ public function getAge()
-        }
+        return [
+            'request' => self::toSerializeableMessage($this->request),
+            'response' => $this->response !== null ? self::toSerializeableMessage($this->response) : null,
+            'staleAt' => $this->staleAt,
+            'staleIfErrorTo' => $this->staleIfErrorTo,
+            'staleWhileRevalidateTo' => $this->staleWhileRevalidateTo,
+            'dateCreated' => $this->dateCreated,
+            'timestampStale' => $this->timestampStale,
+        ];
+    }
+
+    public function __unserialize(array $data): void
+    {
+        $this->request = self::restoreStreamBody($data['request']);
+        $this->response = $data['response'] !== null ? self::restoreStreamBody($data['response']) : null;
+        $this->staleAt = $data['staleAt'];
+        $this->staleIfErrorTo = $data['staleIfErrorTo'];
+        $this->staleWhileRevalidateTo = $data['staleWhileRevalidateTo'];
+        $this->dateCreated = $data['dateCreated'];
+        $this->timestampStale = $data['timestampStale'];
+    }
+
+    /**
+     * Stream/Resource can't be serialized... So we copy the content into an implementation of `Psr\Http\Message\StreamInterface`
+     *
+     * @template T of MessageInterface
+     *
+     * @param T $message
+     * @return T
+     */
+    private static function toSerializeableMessage(MessageInterface $message): MessageInterface
+    {
+        $bodyString = (string)$message->getBody();
 
-        $requestBody = (string)$this->request->getBody();
-        $this->request = $this->request->withBody(
+        return $message->withBody(
             new PumpStream(
-                new BodyStore($requestBody),
+                new BodyStore($bodyString),
                 [
-                    'size' => mb_strlen($requestBody)
+                    'size' => mb_strlen($bodyString),
                 ]
             )
         );
+    }
 
-        return array_keys(get_object_vars($this));
+    /**
+     * @template T of MessageInterface
+     *
+     * @param T $message
+     * @return T
+     */
+    private static function restoreStreamBody(MessageInterface $message): MessageInterface
+    {
+        return $message->withBody(
+            \GuzzleHttp\Psr7\Utils::streamFor((string) $message->getBody())
+        );
     }
 
-    public function __wakeup()
+    public function serialize()
     {
-        // We re-create the stream of the response
-        if ($this->response !== null) {
-            $this->response = $this->response
-                ->withBody(
-                    \GuzzleHttp\Psr7\Utils::streamFor((string) $this->response->getBody())
-                );
-        }
-        $this->request = $this->request
-            ->withBody(
-                \GuzzleHttp\Psr7\Utils::streamFor((string) $this->request->getBody())
-            );
+        return serialize($this->__serialize());
     }
 
+    public function unserialize($data)
+    {
+        $this->__unserialize(unserialize($data));
+    }
```

Comment:
What is the purpose of this function `public function unserialize($data)`? I don't find any call to it and it doesn't seem to be a "magic" PHP function.

Reply:
It's from PHPs legacy system for customized serialization, see: [The Serializable interface](https://www.php.net/manual/en/class.serializable.php)